### ISSUE TITLE
emscripten 2.0.10 causes an assert in mono-threads.c:551

### ISSUE THREAD

**Issue by akoeplinger** [dotnet/runtime#45961]
With 2.0.9 it works but as soon as I update to 2.0.10 I get this:

```
/Users/alexander/dev/runtime/src/mono/mono/utils/hazard-pointer.c:194 <disabled>
/Users/alexander/dev/runtime/src/mono/mono/utils/hazard-pointer.c:194 <disabled>

[... stripped another 50 lines of hazard-pointer.c:194 warnings ...]

==== JS stack trace =========================================

Security context: 0x3c730820fca5 <JSObject>#0#
    0: builtin exit frame: trace(this=0x3c73082056fd <console map = 0x3c7308242915>#1#,0x3c73080e2c81 <String[115]: c"* Assertion at /Users/alexander/dev/runtime/src/mono/mono/utils/mono-threads.c:551, condition `<disabled>' not met\n">,0x3c73082056fd <console map = 0x3c7308242915>#1#)

    1: 556877 [0x3c7308410299] [dotnet.js:1] [bytecode=0x3c7308454855 offset=127](this=0x3c730823f275 <JSGlobal Object>#2#,91757,17007784,4,0,0)
    2: _emscripten_asm_const_int [0x3c73082c99b9] [dotnet.js:1] [bytecode=0x3c73084545a9 offset=33](this=0x3c730823f275 <JSGlobal Object>#2#,556877,5865368,5865344)
    3: WasmToJsFrame [pc: 0x1919dd187918]
    4: WASM [0092209e], function #5748 (''), pc=0x1919dd7f2880 (+0x80), pos=962496 (+77)
    5: WASM [0092209e], function #10909 (''), pc=0x1919dd9bfbfd (+0x9d), pos=1751611 (+32)
    6: WASM [0092209e], function #7150 (''), pc=0x1919dd7fba44 (+0x84), pos=1140976 (+43)
    7: WASM [0092209e], function #3820 (''), pc=0x1919dd5666e8 (+0xc8), pos=597795 (+59)
    8: WASM [0092209e], function #506 (''), pc=0x1919dd2d9226 (+0x66), pos=66433 (+31)
    9: WASM [0092209e], function #229 (''), pc=0x1919dd2f26bc (+0x7c), pos=33489 (+39)
   10: WASM [0092209e], function #102 (''), pc=0x1919dd2fcf16 (+0x36), pos=22751 (+9)
   11: WASM [0092209e], function #6969 (''), pc=0x1919dd874acf (+0x26f), pos=1113722 (+294)
   12: WASM [0092209e], function #2103 (''), pc=0x1919dd65156d (+0x8d), pos=266678 (+39)
   13: WASM [0092209e], function #2740 (''), pc=0x1919dd885505 (+0x65), pos=421771 (+110)
   14: WASM [0092209e], function #7391 (''), pc=0x1919dd69eca4 (+0x304), pos=1170905 (+279)
   15: WASM [0092209e], function #7997 (''), pc=0x1919dd9fbaad (+0x6cd), pos=1274389 (+896)
   16: WASM [0092209e], function #7624 (''), pc=0x1919dd3d7749 (+0x49), pos=1212170 (+12)
   17: WASM [0092209e], function #7187 ('db'), pc=0x1919dd331482 (+0x202), pos=1144461 (+218)
   18: JsToWasmFrame [pc: 0x3c7300084a47]
   19: /* anonymous */ [0x3c73082d1535] [dotnet.js:1] [bytecode=0x3c730845248d offset=46](this=0x3c730823f275 <JSGlobal Object>#2#)
   20: arguments adaptor frame: 2->0
   21: ccall [0x3c73082c82bd] [dotnet.js:1] [bytecode=0x3c73082db34d offset=146](this=0x3c730823f275 <JSGlobal Object>#2#,0x3c7308451b69 <String[22]: #mono_wasm_load_runtime>,0x3c730804216d <null>,0x3c73080a5711 <JSArray[2]>#3#,0x3c73080a5791 <Arguments map = 0x3c7308243d3d>#4#,0x3c7308042309 <undefined>)
   22: /* anonymous */ [0x3c73080a5761] [dotnet.js:1] [bytecode=0x3c73082db16d offset=27](this=0x3c730823f275 <JSGlobal Object>#2#)
   23: arguments adaptor frame: 2->0
   24: _finalize_startup [0x3c730840fba9] [dotnet.js:1] [bytecode=0x3c7308451e3d offset=247](this=0x3c730823f0f5 <Object map = 0x3c7308253e1d>#5#,0x3c7308406451 <Object map = 0x3c7308251645>#6#,0x3c730823fe35 <Object map = 0x3c7308251875>#7#)
   25: onPendingRequestComplete [0x3c730842b225] [dotnet.js:1] [bytecode=0x3c73082de771 offset=52](this=0x3c730823f275 <JSGlobal Object>#2#)
   26: processFetchResponseBuffer [0x3c730823fe85] [dotnet.js:1] [bytecode=0x3c73082ddda1 offset=99](this=0x3c730823f275 <JSGlobal Object>#2#,0x3c7308415161 <Object map = 0x3c73082471e5>#8#,0x3c7308214219 <String[21]: #dotnet.timezones.blat>,0x3c730842af85 <Uint8Array map = 0x3c73082424dd>#9#)
   27: StubFrame [pc: 0x1089185d8]
   28: StubFrame [pc: 0x108881037]
   29: EntryFrame [pc: 0x10885e638]
=====================

* Assertion at /Users/alexander/dev/runtime/src/mono/mono/utils/mono-threads.c:551, condition `<disabled>' not met
```

Repro steps:

```bash
$ ./build.sh -os Browser -c Release
$ cd src/mono/netcore/sample/wasm/console
$ make build
$ make run
```

**Comment by akoeplinger** [comment]
I compared the config.h generated by 2.0.9 and 2.0.10 and they're identical.

/cc @vargaz

**Comment by akoeplinger** [comment]
Turns out this is an emscripten bug/regression: https://github.com/emscripten-core/emscripten/issues/13012

**Comment by lambdageek** [comment]
Fixed in Emscripten 2.0.11
